5-AccessControl/1-call-api-roles/API/TodoListAPI/Startup.cs

```diff
@@ -30,9 +30,36 @@ public void ConfigureServices(IServiceCollection services)
             // This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
             JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
 
-            // Setting configuration for protected web api
+            // Adds Microsoft Identity platform (AAD v2.0) support to protect this Api
             services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
-                .AddMicrosoftIdentityWebApi(Configuration);
+                    .AddMicrosoftIdentityWebApi(options =>
+                    {
+                        options.Events = new JwtBearerEvents();
+
+                        /// <summary>
+                        /// Below you can do extended token validation and check for additional claims, such as:
```

Comment:
>Below [](http://example.com/codeflow?start=28&length=5)

since this API only serves this SPA, you actually need this uncommented and validating that only the clientid of the client app is permitted